packages/office-ui-fabric-react/src/components/ComboBox/VirtualizedComboBox.tsx

```diff
@@ -0,0 +1,44 @@
+import * as React from 'react';
+import { autobind, BaseComponent, customizable } from '../../Utilities';
+import { ComboBox } from './ComboBox';
+import { IComboBoxProps } from './ComboBox.Props';
+import { List } from "../../List";
+import { SelectableOptionMenuItemType, ISelectableOption } from '../../utilities/selectableOption/SelectableOption.Props';
+
+export class VirtualizedComboBox extends BaseComponent<IComboBoxProps, {}> {
+
+  // The virtualized list element
+  private _list: List;
+
+  public render(): JSX.Element {
+    return (
+      <ComboBox
+        {...this.props}
+        onRenderList={ this._onRenderList }
+        onScrollToItem={ this._onScrollToItem }
+      />
+    );
+  }
+
+  @autobind
+  protected _onRenderList(props: IComboBoxProps): JSX.Element {
+    const {
+      onRenderItem
+    } = props;
+
+    // Render virtualized list
+    return (
+      <List
+        componentRef={ this._resolveRef('_list') }
+        role='listbox'
+        items={ props.options }
+        onRenderCell={ onRenderItem != null ? (item: ISelectableOption) => onRenderItem(item) : () => null }
+      />
+    );
+  }
+
+  protected _onScrollToItem(itemIndex: number): void {
+    // We are using the List component, call scrollToIndex
+    this._list.scrollToIndex(itemIndex);
```

Comment:
Does this blow up if itemIndex is null/undefined or is not a valid index (e.g. -1)?

Reply:
I don't believe so, the relevant check is `itemIndex <= index && (itemIndex + itemsPerPage) > index;`, which will always fail if null or negative. It ends up just being a no-op

Reply:
See https://github.com/OfficeDev/office-ui-fabric-react/blob/246e162e1c5573cc597d33e6c1726dbc8c6e457c/packages/office-ui-fabric-react/src/components/List/List.tsx#L179 for the implementation